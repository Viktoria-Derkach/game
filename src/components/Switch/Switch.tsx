import React, { FC, ChangeEvent } from 'react';
import styled from '@emotion/styled';

export interface SwitchProps {
  label: string;
  options: string[];
  value?: string;
  onChange: (event: ChangeEvent<HTMLSelectElement>) => void;
}

export const Switch: FC<SwitchProps> = React.memo(
  ({ label, options, value, onChange }) => (
    <div>
      <Label>{label}: </Label>
      <Select onChange={onChange} value={value}>
        {options.map((option, index) => (
          <option key={index}>{option}</option>
        ))}
      </Select>
    </div>
  )
);

Switch.displayName = 'Switch';

const Label = styled.label`
  margin-right: 0.5rem;
  color: #e6e6e6;
  font-size: 0.85rem;
`;

const Select = styled.select`
  height: 1.6rem;
  padding-left: 0.2rem;
  color: #dddddd;
  background-color: #383b40;
  border: 0.5px solid #010101;
  border-radius: 0.5rem;
  outline: none;
  cursor: pointer;
`;
