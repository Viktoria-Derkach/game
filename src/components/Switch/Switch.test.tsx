import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { Switch } from './Switch';

describe('Switch test cases', () => {
  it('Switch render check', () => {
    const onChange = jest.fn();

    const { asFragment } = render(
      <Switch
        label="Change"
        options={['option1', 'option2', 'option3']}
        onChange={onChange}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
  it('Select level behaviour', () => {
    const onChange = jest.fn();

    render(
      <Switch
        label="Change"
        options={['option1', 'option2', 'option3']}
        onChange={onChange}
      />
    );

    userEvent.selectOptions(screen.getByRole('combobox'), 'option2');

    expect(screen.getByRole('option', { name: 'option2' })).toBeEnabled();

    expect(onChange).toHaveBeenCalled();
  });
});
