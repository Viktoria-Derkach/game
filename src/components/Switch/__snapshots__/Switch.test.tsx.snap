// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch test cases Switch render check 1`] = `
<DocumentFragment>
  <div>
    <label
      class="css-jm2r67"
    >
      Change: 
    </label>
    <select
      class="css-1q4l6vq"
    >
      <option>
        option1
      </option>
      <option>
        option2
      </option>
      <option>
        option3
      </option>
    </select>
  </div>
</DocumentFragment>
`;
